indico/web/client/js/custom_elements/ind_share_widget.js

```diff
@@ -0,0 +1,412 @@
+                    name="server_url"
+                    required
+                    labelPosition="right"
+                    placeholder="https://mastodon.social"
+                    fluid
+                    error={!!submitError && !dirtySinceLastSubmit}
+                    componentLabel={
+                      <Button
+                        icon="save"
+                        positive
+                        type="submit"
+                        disabled={pristine || submitting || (submitFailed && !dirtySinceLastSubmit)}
+                        loading={submitting}
+                        content={Translate.string('Save')}
+                      />
+                    }
+                  />
+                  {submitError && !dirtySinceLastSubmit && (
+                    <div styleName="error-message">
+                      <Icon name="times" />
+                      {submitError}
+                    </div>
+                  )}
+                </form>
+              )}
+            </FinalForm>
+          </>
+        ) : (
+          <>
+            <Message icon positive styleName="saved-mastodon-server-notice">
+              <Icon name="check circle outline" />
+              <Translate>
+                Preferred Mastodon server added! You can now share this event below.
+              </Translate>
+            </Message>
+
+            <Button
+              as="a"
+              target="_blank"
+              href={`${serverURL}/share?text=${shareText}`}
+              icon="share square"
+              content={Translate.string('Share on {mastodonServerName}', {
+                mastodonServerName,
+              })}
+              positive
+              fluid
+              onClick={() => setMastodonOpen(false)}
+            />
+          </>
+        )}
+      </Popup.Content>
+    </Popup>
+  );
+}
+
+SetupMastodonServer.propTypes = {
+  setMastodonOpen: PropTypes.func.isRequired,
+  button: PropTypes.element.isRequired,
+  shareText: PropTypes.string.isRequired,
+};
+
+function ShareWidget({
+  eventName,
+  eventStartDt,
+  eventUrl,
+  shareIcon,
+  googleCalParams,
+  outlookCalParams,
+}) {
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+  const [isMastodonOpen, setMastodonOpen] = useState(false);
+  const [isCopied, setIsCopied] = useState(false);
+  const shareText = `${eventName} (${eventStartDt}) · Indico (${eventUrl})`;
+  return (
+    <Popup
+      trigger={
+        <img
+          src={shareIcon}
+          styleName={isPopupOpen ? 'share-widget show' : 'share-widget'}
+          tabIndex={0}
+          aria-label={Translate.string('Share this page')}
+          aria-modal
+        />
+      }
+      content={
+        <>
+          <div styleName="share-header-title">
+            <Icon name="share alternate" styleName="icon" />
+            <Translate as="strong">Share this page</Translate>
+          </div>
+          <Header size="tiny" styleName="share-section-header">
```

Comment:
Is `size="tiny"` needed?

Reply:
Ah, I probably missed this after playing around with some custom styling.

Removed.